src/aligners/cigar.rs

```diff
@@ -0,0 +1,66 @@
+use std::fmt::Write;
+
+#[derive(PartialEq)]
+pub enum CigarOp {
+    Match,
+    Mismatch,
+    Insertion,
+    Deletion,
+}
+
+impl CigarOp {
+    fn get_char(&self) -> char {
+        match self {
+            CigarOp::Match => 'M',
+            CigarOp::Mismatch => 'X',
+            CigarOp::Insertion => 'I',
+            CigarOp::Deletion => 'D',
+        }
+    }
+
+    fn new(self) -> CigarElement {
+        CigarElement {
+            command: self,
+            length: 1,
+        }
+    }
+}
+
+pub struct CigarElement {
+    command: CigarOp,
+    length: usize,
+}
+
+pub type Cigar = Vec<CigarElement>;
+
+pub trait CigarTrait {
```

Comment:
Ah right. Since `Vec<CigarElement>` is not a type we 'own', we cannot do `impl Cigar` here.

Maybe it's better to not make `Cigar` a type alias, but instead do

```
pub struct Cigar {
  elements: Vec<CigarElement>;
}
```

Then, you don't need the trait and you can just do `impl Cigar` below.